### DISCUSSION TITLE
Chlorine seems to have the wrong atomic weight

### DISCUSSION THREAD

**Discussion by ProfDennis** [rdkit/rdkit#4814] [Development]
See if you reproduce this for the chlorine diatomic molecule. If so, please make a bug report.

MW = ExactMolWt(Chem.MolFromSmiles('ClCl'))
print (MW)

Returns too low a value by about one mass unit.
69.93770536

It should return a value = 70.91

**Comment by greglandrum** [comment]
Hi @ProfDennis,
`ExactMolWt()` returns the exact molecular weight for a molecule: it assumes every atom with an unspecified isotope is in its most common isotopic form and calculates the molecular weight from there.
I think you're looking for "MolWt()", which returns the molecular weight for the molecule using the average isotopic mass for each atom with an unspecfied isotope:
```
In [5]: Descriptors.MolWt(Chem.MolFromSmiles('ClCl'))
Out[5]: 70.906

In [6]: Descriptors.ExactMolWt(Chem.MolFromSmiles('ClCl'))
Out[6]: 69.93770536
```

**Comment by ProfDennis** [comment]
Thanks so much. So much to learn!! I’m really impressed that you responded to this and gave me the answer and the solution. I’m just a week into it with all my other load at the end of the semester, and I am very impressed with this package.
Happy Holidays,
Dennis

On Dec 20, 2021, at 11:09 PM, Greg Landrum ***@***.***> wrote:

﻿

External Email

Hi @ProfDennis<https://github.com/ProfDennis>,
ExactMolWt() returns the exact molecular weight for a molecule: it assumes every atom with an unspecified isotope is in its most common isotopic form and calculates the molecular weight from there.
I think you're looking for "MolWt()", which returns the molecular weight for the molecule using the average isotopic mass for each atom with an unspecfied isotope:

In [5]: Descriptors.MolWt(Chem.MolFromSmiles('ClCl'))
Out[5]: 70.906

In [6]: Descriptors.ExactMolWt(Chem.MolFromSmiles('ClCl'))
Out[6]: 69.93770536


—
Reply to this email directly, view it on GitHub<https://github.com/rdkit/rdkit/discussions/4814#discussioncomment-1850965>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AW3TRKVRBVEUUANNMIO5NELUSAKZPANCNFSM5KPH5B7A>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by mjnoga** [comment]
@greglandrum @ProfDennis 
I believe the confusing part here is that the value returned by ExactMolWt() is actually exact molecular **mass** ("Calculated mass of an ion or molecule with specified isotopic composition." as defined [here](https://www.degruyter.com/document/doi/10.1351/PAC-REC-06-04-06/html). **molecular weight** by definition uses average mass, so weighted by isotope abundances. Therefore when looking at the function names MolWt and ExactMolWt should return the same values. 
To avoid confusion I would recommend renaming ExactMolWt() to ExactMass(), or at least updating the docs.
